File: 881. Boats to Save People/881. Boats to Save People.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    let numberOfBoats = 0
    let leftPtr = 0
    let rightPrt = people.length-1
    people.sort((a,b)=>a-b)

    while(leftPtr<=rightPrt){
        if(people[leftPtr]+people[rightPrt]<=limit){
            leftPtr+=1
            rightPrt-=1
        }else{
            rightPrt-=1
        }

        numberOfBoats+=1
    }

    return numberOfBoats
};

File: 881. Boats to Save People/881. Boats to Save People.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        numberOfBoats = 0
        leftPtr = 0
        rightPtr = len(people)-1
        people.sort()

        while leftPtr <= rightPtr:
            if people[leftPtr] + people[rightPtr] <= limit:
                leftPtr += 1
                rightPtr -= 1
            else:
                rightPtr -= 1
            numberOfBoats += 1

        return numberOfBoats
